kyo-bench/src/main/scala/kyo/bench/TypeMapBench.scala

```diff
@@ -0,0 +1,406 @@
+package kyo.bench
+
+import kyo.TypeMap
+import org.openjdk.jmh.annotations.Benchmark
+import org.openjdk.jmh.annotations.Level
+import org.openjdk.jmh.annotations.OperationsPerInvocation
+import org.openjdk.jmh.annotations.Setup
+import org.openjdk.jmh.infra.Blackhole
+import scala.compiletime.uninitialized
+import zio.ZEnvironment
+
+// Based on: https://github.com/zio/zio/blob/series/2.x/benchmarks/src/main/scala/zio/ZEnvironmentBenchmark.scala
+class TypeMapBench extends Bench(()):
+    import BenchmarkedEnvironment.*
+
+    var zioEnv: ZEnvironment[Env]           = uninitialized
+    var zioEnvSmall: ZEnvironment[SmallEnv] = uninitialized
+    var kyoEnv: TypeMap[Env]                = uninitialized
+    var kyoEnvSmall: TypeMap[SmallEnv]      = uninitialized
+
+    @Setup(Level.Trial)
+    def setup(): Unit =
+        zioEnv = BenchmarkedEnvironment.makeLargeZIO()
+        zioEnvSmall = BenchmarkedEnvironment.makeSmallZIO()
+        kyoEnv = BenchmarkedEnvironment.makeLargeKyo()
+        kyoEnvSmall = BenchmarkedEnvironment.makeSmallKyo()
+    end setup
+
+    @Benchmark
+    def getZIO(bh: Blackhole) =
```

Comment:
What do you think about adding a benchmark measuring the performance of a single `get` as well? It'd be an interesting datapoint.

Reply:
Done.

Quite interesting - the allocation rate seems to be negligible for zio in this case:
```
[info] TypeMapBench.getOneKyo                                   thrpt    5  187410717.151 ± 9570623.077   ops/s
[info] TypeMapBench.getOneKyo:gc.alloc.rate                     thrpt    5          0.010 ±       0.028  MB/sec
[info] TypeMapBench.getOneKyo:gc.alloc.rate.norm                thrpt    5         ≈ 10⁻⁴                  B/op
[info] TypeMapBench.getOneKyo:gc.count                          thrpt    5            ≈ 0                counts
[info] TypeMapBench.getOneZIO                                   thrpt    5  251457709.341 ± 1648544.110   ops/s
[info] TypeMapBench.getOneZIO:gc.alloc.rate                     thrpt    5          0.009 ±       0.020  MB/sec
[info] TypeMapBench.getOneZIO:gc.alloc.rate.norm                thrpt    5         ≈ 10⁻⁴                  B/op
[info] TypeMapBench.getOneZIO:gc.count                          thrpt    5            ≈ 0                counts
```